opm/simulators/flow/FlowProblem.hpp

```diff
@@ -667,7 +667,15 @@ class FlowProblem : public GetPropType<TypeTag, Properties::BaseProblem>
         }
         }
         bool isSubStep = !EWOMS_GET_PARAM(TypeTag, bool, EnableWriteAllSolutions) && !this->simulator().episodeWillBeOver();
-        eclWriter_->evalSummaryState(isSubStep);
+        // For CpGrid with LGRs, ecl/vtk output is not supported yet.
+        const auto& grid =  this->simulator().vanguard().gridView().grid();
+        using GridType =  std::remove_cv_t< typename std::remove_reference<decltype(grid)>::type>;
+        bool isCpGrid = std::is_same_v<GridType, Dune::CpGrid>;
+        if ( isCpGrid && (this->simulator().vanguard().gridView().grid().maxLevel()>0)) {
+        }
+        else {
+            eclWriter_->evalSummaryState(isSubStep);
+        }
```

Comment:
```suggestion
        constexpr bool isCpGrid = std::is_same_v<GridType, Dune::CpGrid>;
        if ( !isCpGrid || (this->simulator().vanguard().gridView().grid().maxLevel()==0)) {
            eclWriter_->evalSummaryState(isSubStep);
        }
```
I hope this suggestion compiles. Empty if branch is  a bit ugly and using `constepr` the comparison will only be done for CpGrid.

Reply:
Thanks for your suggestion